File: app/src/main/java/com/android/app/fancytextreplacer/utils/DialogCustomProgress.java
package com.android.app.fancytextreplacer.utils;


import android.app.Activity;
import android.app.Dialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.Window;
import android.widget.LinearLayout;
import android.widget.ProgressBar;

import com.android.app.fancytextreplacer.R;

public class DialogCustomProgress extends Dialog {
    LinearLayout ll_progress_dialog;
    Activity activity;
    ProgressBar progressBar2;

    public DialogCustomProgress(Activity a) {
        super(a);
        this.activity = a;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        setCanceledOnTouchOutside(false);
        setCancelable(false);
        setContentView(R.layout.custom_progress_dialog_style);
        initUI();
    }

    private void initUI() {
        ll_progress_dialog = (LinearLayout) findViewById(R.id.ll_progress_dialog);
        progressBar2 = (ProgressBar) findViewById(R.id.progressBar2);
        progressBar2.getIndeterminateDrawable().setColorFilter(activity.getResources().getColor(R.color.bottom_nav_tint), android.graphics.PorterDuff.Mode.MULTIPLY);
    }

    public void showProgressDialog() {
        if (this != null) {
            if (this.isShowing()) {
                return;
            }
        }
        show();
    }

    public void dismissProgressDialog() {
        dismiss();
    }

}

File: app/src/main/java/com/stylish/fancy/text/generator/utils/DialogCustomProgress.kt
package com.stylish.fancy.text.generator.utils

import android.app.Activity
import android.app.Dialog
import android.graphics.Color
import android.graphics.PorterDuff
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.View
import android.view.Window
import android.widget.LinearLayout
import android.widget.ProgressBar
import com.stylish.fancy.text.generator.R

class DialogCustomProgress(var activity: Activity) : Dialog(activity) {
    private lateinit var progressDialog: LinearLayout
    private lateinit var progressBar: ProgressBar

    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        setCanceledOnTouchOutside(false)
        setCancelable(false)
        setContentView(R.layout.custom_progress_dialog_style)
        initUI()
    }

    private fun initUI() {
        progressDialog = findViewById<View>(R.id.ll_progress_dialog) as LinearLayout
        progressBar = findViewById<View>(R.id.progressBar2) as ProgressBar
        progressBar.indeterminateDrawable.setColorFilter(
            activity.resources.getColor(R.color.bottom_nav_tint),
            PorterDuff.Mode.MULTIPLY
        )
    }

    fun dismissProgressDialog() {
        dismiss()
    }
}